### PR TITLE
used f-strings in docs

### PR DESCRIPTION
**Pull request by PSY27**
- [x] xref #32039
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.10.0.rst [hunk 1]
```diff
@@ -297,7 +297,7 @@ Updated PyTables support
 
 .. code-block:: ipython
 
-    In [41]: store = pd.HDFStore('store.h5')
+    In [41]: store = pd.HDFStore('store.h5')
```
Comment (rohitsanj) [inline doc/source/whatsnew/v0.10.0.rst]: undo this change, remove the whitespace

File: doc/source/user_guide/io.rst [hunk 2]
```diff
@@ -3807,14 +3807,14 @@ The right-hand side of the sub-expression (after a comparison operator) can be:
 
    .. code-block:: python
 
-      string = "HolyMoly'"
+      string = "HolyMoly"
       store.select('df', 'index == string')
 
    instead of this
```
Comment (rohitsanj) [inline doc/source/user_guide/io.rst]: please remove this line

**Review by rohitsanj — CHANGES_REQUESTED** [review]
some comments for your consideration

File: doc/source/whatsnew/v0.10.0.rst (hunk 1 above)
Comment (PSY27) [reply doc/source/whatsnew/v0.10.0.rst]: ok
thanks for the input

**Comment by WillAyd** [comment]
I think this is correct as is - isn't this supposed to show how the single quote doesn't get escaped? See text immediately following

File: doc/source/user_guide/io.rst [hunk 3]
```diff
@@ -3807,14 +3807,12 @@ The right-hand side of the sub-expression (after a comparison operator) can be:
...
+      string = "HolyMoly"
       store.select('df', 'index == string')
 
-   instead of this
-
    .. code-block:: ipython
 
-      string = "HolyMoly'"
+      string = "HolyMoly"
       store.select('df', 'index == %s' % string)
```
Comment (WillAyd) [inline doc/source/user_guide/io.rst]: You can update this to use an f-string if you'd like to change something

**Review by WillAyd — CHANGES_REQUESTED** [review]
as mentioned let's revert the change to the string, but can use f-strings if you'd like to make a change

**Comment by MarcoGorelli** [comment]
> I think this is correct as is - isn't this supposed to show how the single quote doesn't get escaped? See text immediately following

Yup, my bad - sorry @PSY27 . Though as Will said, if you change line 3818 to use an f-string, that would be good

File: doc/source/user_guide/io.rst (hunk 3 above)
Comment (rohitsanj) [inline doc/source/user_guide/io.rst]: Change this to 
```python
store.select('df', f'index == {string}')
```

File: doc/source/user_guide/io.rst [hunk 4]
```diff
@@ -3807,14 +3807,12 @@ The right-hand side of the sub-expression (after a comparison operator) can be:
 
    .. code-block:: python
 
-      string = "HolyMoly'"
+      string = "HolyMoly"
```
Comment (rohitsanj) [inline doc/source/user_guide/io.rst]: Revert this change

File: doc/source/user_guide/io.rst [hunk 5]
```diff
@@ -3807,14 +3807,12 @@ The right-hand side of the sub-expression (after a comparison operator) can be:
...
       store.select('df', 'index == string')
 
-   instead of this
-
    .. code-block:: ipython
 
-      string = "HolyMoly'"
+      string = "HolyMoly"
```
Comment (rohitsanj) [inline doc/source/user_guide/io.rst]: Revert this change as well

File: doc/source/user_guide/io.rst [hunk 6]
```diff
@@ -3807,14 +3807,12 @@ The right-hand side of the sub-expression (after a comparison operator) can be:
 
    .. code-block:: python
 
-      string = "HolyMoly'"
+      string = "HolyMoly"
       store.select('df', 'index == string')
```
Comment (rohitsanj) [inline doc/source/user_guide/io.rst]: @WillAyd shouldn't this be changed to use an f-string too?

**Review by rohitsanj — CHANGES_REQUESTED** [review]

**Comment by PSY27** [comment]
@WillAyd @MarcoGorelli  please review it now

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Looks good, thanks @PSY27!

While we're here, there's some other `.format`s which could be changed within this file - for example, this one on line 3828:

```
store.select('df', f'index == {string!r}')
```

Else I think it's fine as is, I realise the requirement for this PR wasn't clear upfront :)

EDIT
----

Though if you do make this extra change (line 3828) please also change the line
```
If you *must* interpolate, use the ``'%r'`` format specifier
```
to
```
If you *must* interpolate, use the ``!r`` conversion flag
```

**Comment by jreback** [comment]
thanks @PSY27

**Pull request merged**